src/store/index-level.ts

```
@@ -1,213 +1,435 @@
+   *
+   * @param filters Array of filters that are treated as an OR query.
+   * @param queryOptions query options for sort and pagination, requires at least `sortProperty`. The default sort direction is ascending.
+   * @param options IndexLevelOptions that include an AbortSignal.
+   * @returns {string[]} an array of itemIds that match the given filters.
+   */
+  async query(tenant: string, filters: Filter[], queryOptions: QueryOptions, options?: IndexLevelOptions): Promise<string[]> {
+    // returns an array of which search filters we need to perform a full search on.
+    // if there are no search filters returned, we do a full scan on the sorted index.
+    const searchFilters = FilterSelector.select(filters, queryOptions);
+    if (searchFilters.length === 0) {
+      return this.sortedIndexQuery(tenant, filters, queryOptions, options);
+    }
+
+    return this.filteredIndexQuery(tenant, filters, searchFilters, queryOptions, options);
+  }
+
+  /**
+   * Queries the sort property index for items that match the filters. If no filters are provided, all items are returned.
+   * This query is a linear iterator over the sorted index, checking each item for a match.
+   * If a cursor is provided it starts the iteration at the cursor point.
+   */
+  async sortedIndexQuery(tenant: string, filters: Filter[], queryOptions: QueryOptions, options?: IndexLevelOptions): Promise<string[]> {
+    const { limit, cursor , sortProperty } = queryOptions;
+    const matches: string[] = [];
+
+    // if there is a cursor we fetch the starting key given the sort property, otherwise we start from the beginning of the index.
+    const startKey = cursor ? await this.getStartingKeyForCursor(tenant, cursor, sortProperty, filters) : '';
+    if (startKey === undefined) {
+      // getStartingKeyForCursor returns undefined if an invalid cursor is provided, we return an empty result set.
```

Should we throw an error instead of resulting empty array?